boards/arm/teensy4/teensy41.dts

```diff
@@ -5,18 +5,29 @@
  */
 
 #include "teensy40.dts"
+/ {
+	chosen {
+		zephyr,flash-controller = &w25q64jvxgim;
+		zephyr,flash = &w25q64jvxgim;
+	};
+};
 
 /delete-node/ &w25q16jvuxim;
 &flexspi {
-	reg = < 0x402a8000 0x4000 >, < 0x60000000 0x800000 >;
+	status = "okay";
+
+	reg = <0x402a8000 0x4000>, <0x60000000 0x800000>;
 	/* WINBOND flash memory*/
 	w25q64jvxgim: w25q64jvxgim@0 {
-		compatible = "winbond,w25q64jvxgim";
-		size = < 8388607 >;
-		reg = < 0 >;
-		spi-max-frequency = < 133000000 >;
+		compatible = "nxp,imx-flexspi-nor";
+		size = <67108864>;
+		reg = <0>;
+		spi-max-frequency = <133000000>;
```

Comment:
This could use `DT_FREQ_M(133)`, again not quite an explicit rule but makes it easier to understand imo.